### PR TITLE
Mon 15007 theme modification for 22 10 release

### PR DESCRIPTION
**Pull request by PascalTassel**
## Description
Rework the 4 following things necessary for 22.10 release:

- Make badges in top counter bigger
- Change panel background color
- Change menu background color
- Display separator under Resources satatus  search bar to differentiate visualy the detail panel from the rest

Extras:

- Change page background color
- Set header height to 64px and adjust header submenus positon Y

## Type of change

lightMode

darkMode

- [X] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [X] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

1. Change **centreon-frontend** package url  into the centreon `package.json` file replacing `develop` by the branch name :
```
"centreon-frontend": "git+https://centreon@github.com/centreon/centreon-frontend.git#MON-15007-theme-modification-for-22-10-release
```
2. Update package into `centreon` app:
```
npm update centreon-frontend --legacy-peer-deps
```
3. Restart `centreon` app from `centreon-dev-ecosystem`:
```
pm2 restart centreon
```

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by Wedierre — COMMENTED** [review]
to follow up on our update this morning on interfaces:
I should think about incorporating the panel and sub-menu changes for the Dark and White theme.

Namely:
Integrate new colors for Dark and White submenus
https://xd.adobe.com/view/7d7eba4e-e65a-45a7-bcf6-40953233e3a7-282d/
Sub-menu Dark.pngSous-menu White.png
Use the same background for the Panel as for the sub-menus.
Check the behavior of the color information bubbles in the case of 3-digit data. (see together)
Integrate the overview (color and change of pictograms) and keep the same state enabled when the heading or one of its sub-headings is selected for navigation pictograms (aside left)


I think that by integrating if possible these modifications will allow our users to better navigate our interfaces (to be tested and improved following feedback)
Sous-menu Dark
Sous-menu White

**Comment by bdauria** [comment]
@PascalTassel can you update the screens as well as adding the missing ones?

File: www/front_src/src/Header/PollerMenu/PollerStatusIcon/index.tsx [hunk 1]
```diff
@@ -70,8 +66,8 @@ const useStatusStyles = makeStyles<Theme, StyleProps>((theme) => {
     database: ({ databaseSeverity }): CreateCSSProperties<StyleProps> =>
       getSeverityColor(databaseSeverity),
     icon: {
-      height: theme.spacing(1.5),
-      width: theme.spacing(1.5),
+      height: '14px',
+      width: '14px',
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/PollerStatusIcon/index.tsx]: We prefer to use the spacing unit system from MUI

File: www/front_src/src/Header/PollerMenu/index.tsx [hunk 2]
```diff
@@ -111,11 +110,11 @@ const useStyles = makeStyles((theme) => ({
       : theme.palette.primary.main,
     boxSizing: 'border-box',
     display: 'none',
-    left: theme.spacing(0),
+    left: 0,
     padding: theme.spacing(1),
     position: 'absolute',
     textAlign: 'left',
-    top: '100%',
+    top: 'calc(100% + 9px)',
```
Comment (Thebarda) [inline www/front_src/src/Header/PollerMenu/index.tsx]: Use spacing unit

File: www/front_src/src/Header/SwitchThemeMode/index.tsx [hunk 3]
```diff
@@ -40,8 +38,7 @@ const useStyles = makeStyles((theme) => ({
     },
   },
   disabledMode: {
-    color: theme.palette.common.white,
-    opacity: 0.5,
+    opacity: '.5',
```
Comment (Thebarda) [inline www/front_src/src/Header/SwitchThemeMode/index.tsx]:
```suggestion
    opacity: 0.5,
```

File: www/front_src/src/Header/userMenu/index.tsx [hunk 4]
```diff
@@ -131,53 +112,53 @@ const useStyles = makeStyles((theme) => ({
...
+    },
+    padding: 0,
   },
   menu: {
-    backgroundColor: equals(theme.palette.mode, ThemeMode.dark)
-      ? theme.palette.background.default
-      : theme.palette.primary.main,
+    // backgroundColor: equals(theme.palette.mode, ThemeMode.dark)
+    //   ? theme.palette.background.default
+    //   : theme.palette.primary.main,
```
Comment (Thebarda) [inline www/front_src/src/Header/userMenu/index.tsx]: Remove if not necessary

File: www/front_src/src/Header/userMenu/index.tsx [hunk 5]
```diff
@@ -131,53 +112,53 @@ const useStyles = makeStyles((theme) => ({
...
+    backgroundColor: theme.palette.background.paper,
     border: 'none',
     borderRadius: 0,
-    color: theme.palette.common.white,
+    fontSize: '.75rem',
```
Comment (Thebarda) [inline www/front_src/src/Header/userMenu/index.tsx]: Is it possible to use a fontSize from a MUI typography (body1, body2, caption, etc...) ?

File: www/front_src/src/Main/MainLoader.tsx [hunk 6]
```diff
@@ -11,7 +11,7 @@ import { labelCentreonIsLoading } from './translatedLabels';
 const useStyles = makeStyles((theme) => ({
   loader: {
     alignItems: 'center',
-    backgroundColor: theme.palette.background.paper,
+    backgroundColor: theme.palette.background.default,
```
Comment (Thebarda) [inline www/front_src/src/Main/MainLoader.tsx]:
```suggestion
    backgroundColor: theme.palette.background.paper,
```

File: www/front_src/src/App/index.tsx [hunk 7]
```diff
@@ -34,13 +34,14 @@ const useStyles = makeStyles((theme) => ({
...
   },
   mainContent: {
-    backgroundColor: theme.palette.background.paper,
-    height: `calc(100vh - ${theme.spacing(headerHeight + headerPadding)})`,
+    height: `calc(100vh - ${theme.spacing(headerHeight)})`,
     width: '100%',
   },
   wrapper: {
     alignItems: 'stretch',
+    backgroundColor: theme.palette.background.default,
```
Comment (Thebarda) [inline www/front_src/src/App/index.tsx]:
```suggestion
    backgroundColor: theme.palette.background.paper,
```

Comment (PascalTassel) [reply www/front_src/src/App/index.tsx]: I don't understand : white is the default background color for light theme, paper background (#f4f4f4) is used for panels, submenus, etc.

File: www/front_src/src/App/index.tsx [hunk 8]
```diff
@@ -34,13 +35,16 @@ const useStyles = makeStyles((theme) => ({
...
   mainContent: {
-    backgroundColor: theme.palette.background.paper,
-    height: `calc(100vh - ${theme.spacing(headerHeight + headerPadding)})`,
+    height: `calc(100vh - ${theme.spacing(headerHeight)})`,
     width: '100%',
   },
   wrapper: {
     alignItems: 'stretch',
+    backgroundColor: equals(theme.palette.mode, ThemeMode.dark)
+      ? '#2c2c2c'
```
Comment (Thebarda) [inline www/front_src/src/App/index.tsx]: This must use a color from the theme

Comment (PascalTassel) [reply www/front_src/src/App/index.tsx]: I replaced hexa code with theme.palette.background.paper (as light theme).

**Review by Thebarda — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**